lib/galaxy/util/path/__init__.py

```diff
@@ -343,6 +345,26 @@ def canonicalize(self, ext):
 })
 
 
+def external_chown(path, pwent, external_chown_script, description="file"):
+    """
+    call the external chown script (if not None) to change
+    the user and group of the given path, and additional description
+    of the file/path for the log message can be given
+    """
+    if external_chown_script is None:
+        return
+
+    try:
```

Comment:
You could use commands.execute here, right? (https://github.com/galaxyproject/galaxy/pull/9544)

Reply:
> Is this change critical for fixing hisitory import/export ? If not I would keep the current behavior and change it in another PR.

Could be a good idea. But then `external_chown` needs a proper return value: maybe `True` for success and `False` for failed executions or `external_chown_script is None`